annotations/generate_ld_data.py

```diff
@@ -0,0 +1,165 @@
+from gnomad_hail import *
+from hail.utils import new_temp_file
+from hail.utils.java import Env
+from hail.linalg import BlockMatrix
+
+
+def ld_matrix_path(data_type: str, pop: str, version: str = CURRENT_RELEASE):
+    return f'gs://gnomad-resources/ld/gnomad.{data_type}.r{version}.{pop}.ld.bm'
+
+
+def ld_index_path(data_type: str, pop: str, version: str = CURRENT_RELEASE):
+    return f'gs://gnomad-resources/ld/gnomad.{data_type}.r{version}.{pop}.ld.variant_indices.ht'
+
+
+def ld_pruned_path(data_type: str, pop: str, r2: str, version: str = CURRENT_RELEASE):
+    return f'gs://gnomad-resources/ld/gnomad.{data_type}.r{version}.{pop}.ld.pruned_set.r2_{r2}.ht'
+
+
+def ld_scores_path(data_type: str, pop: str, version: str = CURRENT_RELEASE):
+    return f'gs://gnomad-resources/ld/gnomad.{data_type}.r{version}.{pop}.ld_scores.ht'
+
+
+eur_w_ld_chr_ht = 'gs://konradk/ld_score/reference_data/eur_w_ld_chr/20.l2.ldscore.ht'
+eur_ref_ld_chr_ht = 'gs://konradk/ld_score/reference_data/eur_ref_ld_chr/20.l2.ldscore.ht'
+kg_ld_ht = 'gs://konradk/ld_score/reference_data/1000G_Phase3_baselineLD_v2.1_ldscores/baselineLD.20.l2.ldscore.ht'
+
+
+def get_pop_and_subpop_counters(mt):
+    cut_dict = {'pop': hl.agg.filter(hl.is_defined(mt.meta.pop), hl.agg.counter(mt.meta.pop)),
+                'subpop': hl.agg.filter(hl.is_defined(mt.meta.subpop), hl.agg.counter(mt.meta.subpop))
+                }
+    cut_data = mt.aggregate_cols(hl.struct(**cut_dict))
+    return cut_data
+
+
+def filter_mt_for_ld(mt, label, pop):
+    pop_mt = mt.filter_cols(mt.meta[label] == pop)
+    meta_index = hl.zip_with_index(pop_mt.freq_meta).find(lambda f: (f[1].get(label) == pop)).collect()[0][0]
+    pop_mt = pop_mt.annotate_rows(pop_freq=pop_mt.freq[meta_index])
+    pop_mt = pop_mt.filter_rows((hl.len(pop_mt.filters) == 0) &
+                                (pop_mt.pop_freq.AC > 1) &
+                                (pop_mt.pop_freq.AN - pop_mt.pop_freq.AC > 1) &
+                                ~((pop_mt.pop_freq.AF == 0.5) &
+                                  (pop_mt.pop_freq.homozygote_count == 0)))
+    return pop_mt
+
+
+def import_reference_ld_scores():
```

Comment:
This is never called -- is it meant to be imported then used? If so is that the best place for it?